src/Grpc/JsonTranscoding/src/Microsoft.AspNetCore.Grpc.JsonTranscoding/Internal/Json/JsonConverterHelper.cs

```diff
@@ -36,12 +37,16 @@ internal static JsonSerializerOptions CreateSerializerOptions(JsonContext contex
         // For streaming to work, indenting must be disabled when streaming.
         var writeIndented = !isStreamingOptions ? context.Settings.WriteIndented : false;
 
+        var typeInfoResolver = JsonTypeInfoResolver.Combine(
```

Comment:
You might want to consider using the `Modifiers` property on `DefaultJsonTypeInfoResolver`, which lets you register callbacks that mutate the default metadata.

Reply:
I don't care about any reflection metadata. Creating the type info from scratch seems better in this situation.